app/javascript/src/ChangePassword/components/ChangePasswordHooks.jsx

```diff
@@ -1,87 +1,103 @@
+// @flow
+
 import { useState } from 'react'
-import { validateForm, validateSingleField } from 'LoginPage/utils/formValidation'
+import validate from 'validate.js' // TODO: including this since validateSingleField and validateForm from LoginPage/utils/formValidation screams for a refactor
+
+type PasswordField = 'user[password]'
+type PasswordConfirmationField = 'user[password_confirmation]'
+type FieldName = PasswordField | PasswordConfirmationField
+type ValidatorName = 'presence' | 'length' | 'equality'
+type ValidatorOption = {message: string, minimun?: number, attribute?: string}
+type Constraints = {[FieldName]: {[ValidatorName]: ValidatorOption}}
+type FieldState = {[FieldName]: string}
+type ValidationErrors = {[FieldName]: string[]}
```

Comment:
This should be an optional type, since validate can return nothing (undefined).

Reply:
`validate` fn might return ValidationErrors or undefined, better using it in the fn call not the type definition